bin/dev_env_menu

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+# This script provides a menu of Service Broker tests to be run.
+# These tests are intended to be run within the Service Broker test
+# container that is created as part of the development environment
+# (via the 'start' script).
+
+# Run this from the root project directory so that rspec and Cucumber
+# tests can be discovered.
+cd "$(dirname "$0")/.." || (echo "Could not cd to parent dir"; exit 1)
+
+function main_menu() {
+
+  local options=("Run rspec unit tests" \
+           "Run integration (non-E2E) Cucumber tests" \
+           "Select from a list of Cucumber features to test" \
+           "Select from a list of Cucumber scenarios to test" \
+           "Run a bash shell in test container" \
+           "Exit and clean up development environment" )
+
+  while true; do
+    PS3=$'\nPlease select tests to run: '
+    echo
+    select option in "${options[@]}"; do
+      case $REPLY in
+          1) unit_test; break ;;
+          2) integration_test; break ;;
+          3) feature_menu; break ;;
+          4) scenario_menu; break ;;
+          5) bash; break ;;
+          6) exit 0
+      esac
+    done
+  done
+}
+
+function feature_menu() {
+  PS3=$'\nPlease select a feature to test: '
+  sub_menu "Feature: "
+}
+
+function scenario_menu() {
+  PS3=$'\nPlease select a scenario to test: '
+  sub_menu "Scenario: "
+}
+  
+function sub_menu() {
+  # Search for instances of the specified pattern in Cucumber dry run output
+  pattern="$1"
+  IFS=$'\n'
+  local options=( $(dry_run | grep "$pattern" | sed "s/[ ]*$pattern[ ]*//") )
+  unset IFS
+  
+  # Add an option to return to the main menu
+  options+=("Return to main menu")
+  num_opts="${#options[@]}"
+  
+  while true; do
+    echo
+    select option in "${options[@]}"; do
+      if [ "$REPLY" -eq "$num_opts" ]; then
+        break 2;
+      elif [ "$REPLY" -ge 1 ] && [ "$REPLY" -le $((num_opts-1)) ]; then
+        test_by_name "$option"
+      fi
+      break;
+    done
+  done
+}
+
+function dry_run() {
+  cucumber --dry-run --tags ~@integration --format pretty --quiet
```

Comment:
(Nit) I'd add double quotes around "~@integration"

Reply:
D'oh. Thanks for catching that!